role-make-version-changelog.sh

```diff
@@ -114,91 +87,122 @@ else
 fi
 echo Using branch "$BRANCH" for the changelog commit/PR
 
-# get latest tag
-latest_tag=$(git describe --tags --abbrev=0 2> /dev/null || :)
-# special case for network and sshd
-allow_v=""
-case "$latest_tag" in
-v*)
-    if [ "$repo" = ansible-sshd ] || [ "$repo" = sshd ]; then
-        # sshd uses a leading v
-        allow_v=v
-    else
-        # network had a case where there where two tags for the
-        # latest - one with leading v and one without - git describe
-        # would return the one with the leading v - so had to strip
-        # it off
-        latest_tag="${latest_tag//v}"
-    fi
-    ;;
-esac
+# get latest release
+releases_latest=$(gh api repos/"$owner"/"$repo"/releases/latest -q '.tag_name,.published_at' 2> /dev/null || :)
```

Comment:
```suggestion
read -r latest_tag release_timestamp <<< $(gh api repos/"$owner"/"$repo"/releases/latest -q '.tag_name + " " + .published_at' 2> /dev/null || :)
```
if the release exists:
```yaml
read -r latest_tag release_timestamp <<< $(gh api repos/linux-system-roles/network/releases/latest -q '.tag_name + " " + .published_at' 2>&1 || :)
echo $latest_tag
1.12.0
echo $release_timestamp
2023-05-30T19:34:49Z
```
if there is no release
```
read -r latest_tag release_timestamp <<< $(gh api repos/linux-system-roles/bootloader/releases/latest -q '.tag_name + " " + .published_at' 2>&1 || :)
echo $latest_tag
{"message":"Not
echo $release_timestamp
Found","documentation_url":"https://docs.github.com/rest/reference/repos#get-the-latest-release"}gh: Not Found (HTTP 404)
```
so you could still use a similar comparison for `if [[ "$release_timestamp" == *"Not Found"* ]]; then`

Reply:
I'd need too look for `message":"Not` or `Found",` in `$latest_tag` or `$release_timestamp`, isn't it uglier than the initial solution? With `-q '.tag_name,.published_at'`, it prints

Reply:
ok - then leave it - it's fine